guarddog/analyzer/metadata/pypi/repository_integrity_missmatch.py

```diff
@@ -0,0 +1,242 @@
+""" Empty Information Detector
+
+Detects if a package contains an empty description
+"""
+import configparser
+import hashlib
+import os
+import re
+from typing import Optional, Tuple
+
+import pygit2  # type: ignore
+import urllib3.util
+
+from guarddog.analyzer.metadata.repository_integrity_missmatch import IntegrityMissmatch
+
+GH_REPO_REGEX = r'(?:https?://)?(?:www\.)?github\.com/(?:[\w-]+/)(?:[\w-]+)'
+GH_REPO_OWNER_REGEX = r'(?:https?://)?(?:www\.)?github\.com/([\w-]+)/([\w-]+)'
+
+
+def extract_owner_and_repo(url) -> Tuple[Optional[str], Optional[str]]:
+    match = re.search(GH_REPO_OWNER_REGEX, url)
+    if match:
+        owner = match.group(1)
+        repo = match.group(2)
+        return owner, repo
+    return None, None
+
+
+def find_best_github_candidate(cand, name):
```

Comment:
maybe a quick comment on what the function does? and possibly more descriptive variable names than `cand`